src/core/utils/FocusTrap.tsx

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { getFocusableElements } from './common';
+import { mergeRefs } from './hooks/useMergedRefs';
+
+export type FocusTrapProps = {
+  /**
+   * A single child element to trap focus in.
+   */
+  children: JSX.Element;
+};
+
+/**
+ * Trap focus within child container.
+ * Used for modals and dialogs.
+ */
+export const FocusTrap = (props: FocusTrapProps) => {
+  const { children } = props;
+  const childRef = React.useRef<HTMLElement>();
+
+  const onLastFocus = (event: React.FocusEvent) => {
+    const elements = getFocusableElements(childRef.current);
+    const firstElement = elements[0];
+    const lastElement = elements[(elements.length || 1) - 1];
+
+    if (event.relatedTarget === lastElement) {
+      (firstElement as HTMLElement).focus();
+    } else {
+      (lastElement as HTMLElement).focus();
+    }
+  };
+
+  const onFirstFocus = (event: React.FocusEvent) => {
+    const elements = getFocusableElements(childRef.current);
+    const firstElement = elements[0];
+    const lastElement = elements[(elements.length || 1) - 1];
+
+    if (
+      event.relatedTarget === childRef.current ||
+      event.relatedTarget === firstElement
+    ) {
+      (lastElement as HTMLElement).focus();
+    } else {
+      (firstElement as HTMLElement).focus();
+    }
+  };
+
+  return (
+    <div>
+      <div tabIndex={0} onFocus={onFirstFocus} />
```

Comment:
So I was hoping that these two divs would simplify the implementation and you wouldn't need to find all focusable elements.

Since that is not the case, maybe just use first/last elements to loop focus? I imagine you'd have to add `onKeyDown` (or maybe `onBlur`). https://developers.google.com/web/fundamentals/accessibility/focus/using-tabindex#modals_and_keyboard_traps


There is also an accessibility benefit to removing the empty divs, because tabbable elements are announced by screen readers and should not be hidden.

Reply:
I had it done, none of you liked it. Code is less nice.
![image](https://user-images.githubusercontent.com/81580355/130740734-e2966601-c801-4d10-a89c-cce16d49d5bd.png)
It was with hook, but pretty much same..

Reply:
I added hook now too..
Could you please have a look and cast your vote: Hook or Component.
@bentleyvk

Reply:
My vote goes to.... divs. It looks better and easier to understand. Just add `aria-hidden` to those divs and it should be ok with those screen readers.

Reply:
`aria-hidden` on focusable elements is considered a bad practice for both accessibility and SEO

https://www.w3.org/TR/using-aria/#fourth
https://web.dev/aria-hidden-focus/